src/arrayMethodJoin.js

```diff
@@ -4,9 +4,44 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    // if input array is empty - return empty string
+    if (this.length === 0) {
+      return '';
+    }
+
+    // we will process input separator
+    let fixedSeparator = separator;
+
+    // process separator: if it is null - change it to string 'null'
+    if (separator === null) {
+      fixedSeparator = 'null';
+    }
+
+    // convert input separator to string
+    fixedSeparator = fixedSeparator.toString();
```

Comment:
I tested your code and everything works fine without this piece of code:
![image](https://user-images.githubusercontent.com/36279782/190452579-6864946c-6c61-4276-a1ca-c57bbd095669.png)